File: Tareas/tarea14.java
import javax.swing.*;

public class Tarea14 {
    public static void main(String[] args) {
        JOptionPane.showMessageDialog(null, "Azul Carolina Martínez Cabañas\n1868435 ITS\nPeriodo Enero-Junio 2020\nIng. Ismael Gutiérrez Pimentel");
        int n = Integer.parseInt(JOptionPane.showInputDialog("Búsqueda en Matriz\nIngrese la cantidad de filas de la matriz: "));
        int m = Integer.parseInt(JOptionPane.showInputDialog("Ingrese la cantidad de columnas de la matriz. "));
        int[][] matriz = new int[n][m];
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                matriz[i][j] = Integer.parseInt(JOptionPane.showInputDialog("Ingrese un número a la matriz"));
            }
        }
        String matrizImpresa = "";
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                matrizImpresa += matriz[i][j];
                matrizImpresa += " ";
            }
            matrizImpresa += "\n";
        }
        JOptionPane.showMessageDialog(null, matrizImpresa);
        int buscar = Integer.parseInt(JOptionPane.showInputDialog("Ingrese el número a buscar"));
        int contador = 0;
        for (int i = 0; i < n; i++) {
            for (int j = 0; j < m; j++) {
                if (matriz[i][j] == buscar) {
                    contador += 1;
                }
            }
        }
        String impresion = "El número a buscar es: " + buscar;
        if (contador > 0) {
            impresion += "\nVeces que aparece en la matriz: " + contador;
        } else {
            impresion += "\nNo se encontró al numero en la matriz";
        }
        JOptionPane.showMessageDialog(null, impresion);
    }
}

File: Tareas/tarea14.rb
puts "Azul Carolina Martínez Cabañas"
puts "1868435					 ITS"
puts "Periodo Enero-Junio 2020"
puts "Ing. Ismael Gutiérrez Pimentel"
puts "Búsqueda en Matriz"
puts "Ingrese la cantidad de filas de la matriz: "
n = gets().to_i
puts "Ingrese la cantidad de columnas de la matriz: "
m = gets().to_i
matriz= Array.new(n){Array.new(m)}
for i in (0...n)
    for j in (0...m)
    puts "Ingrese el número a ingresar en la matriz: \n"
    matriz[i][j] = gets.to_i
    end
end

for i in (0...n)
    for j in (0...m)
        print "#{matriz[i][j]}   "
    end
    puts "\n"
end
puts "Ingrese el número a buscar"
buscar = gets().to_i
contador = 0
for i in (0...n)
    for j in (0...n)
    if(matriz[i][j] == buscar)
        contador += 1
    end
    end
end
puts "\nEl numero a buscar es: #{buscar}"

if contador>1
    puts "\nVeces que aparece en el matriz: #{contador}"
else
    puts "No se encontró al número en la matriz"
end
